adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRewardedAdRenderer.m

```diff
@@ -0,0 +1,133 @@
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+        
+    _rewardedAd = [MTGBidRewardAdManager sharedInstance];
+    [_rewardedAd loadVideoWithBidToken:_adConfiguration.bidResponse placementId:placementId unitId:adUnitId delegate:self];
+}
+
+#pragma mark MTGRewardAdLoadDelegate
+- (void)onVideoAdLoadSuccess:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    if (_adLoadCompletionHandler) {
+        _adEventDelegate = _adLoadCompletionHandler(self, nil);
+    }
+}
+
+- (void)onVideoAdLoadFailed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId error:(nonnull NSError *)error {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil, error);
+    }
+}
+
+#pragma mark MTGRewardAdShowDelegate
+- (void)onVideoAdShowSuccess:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate willPresentFullScreenView];
+    [_adEventDelegate reportImpression];
+    [_adEventDelegate didStartVideo];
+}
+
+- (void)onVideoAdShowFailed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId withError:(nonnull NSError *)error {
+    [_adEventDelegate didFailToPresentWithError:error];
+}
+
+- (void)onVideoPlayCompleted:(nullable NSString *)placementId unitId:(nullable NSString *)unitId{
+    [_adEventDelegate didEndVideo];
+}
+
+- (void)onVideoAdClicked:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate reportClick];
+}
+
+- (void)onVideoAdDismissed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId withConverted:(BOOL)converted withRewardInfo:(nullable MTGRewardAdInfo *)rewardInfo {
+    [_adEventDelegate willDismissFullScreenView];
+    if (converted) {
+        [_adEventDelegate didRewardUser];
+    }
+}
+
+- (void)onVideoAdDidClosed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate didDismissFullScreenView];
+}
+
+#pragma mark GADMediationRewardedAd
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    NSString *adUnitId = _adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = _adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    GADMAdapterMintegralExtras *extras = _adConfiguration.extras;
+    _rewardedAd.playVideoMute = extras.playVideoMute;
+    if ([_rewardedAd isVideoReadyToPlayWithPlacementId:placementId unitId:adUnitId]) {
+        [_rewardedAd showVideoWithPlacementId:placementId unitId:adUnitId withRewardId:nil userId:nil delegate:self viewController:viewController];
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdNotValid, @"Unable to display ad.");
```

Comment:
Nit, update the comment to: `Failed to display rewarded video ad from Mintegral.`

Reply:
ok,i have updated.